### PR TITLE
Testsuite: fix gmsh tests

### PR DESCRIPTION
**Pull request by tamiko**
- add MPI_InitFinialize to the api tests so that the MPI context is properly initalized and finalized. Otherwise our tests fail when gmsh is compiled with MPI support.
 - add output variants for the two unstable tests.

In reference to #15383

### CODE REVIEW

File: tests/gmsh/gmsh_api_01.cc [hunk 1]
```diff
@@ -22,8 +22,10 @@
 #include "../tests.h"
 
 int
-main()
+main(int argc, char **argv)
 {
+  // gmsh might be build with mpi support enabled.
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv);
```
Comment (bangerth) [inline tests/gmsh/gmsh_api_01.cc:28]: Why is this necessary? These tests aren't run under MPI, are they?

Comment (tamiko) [reply tests/gmsh/gmsh_api_01.cc:28]: @bangerth When you build gmsh with MPI support then gmsh is running under an MPI context.
Test failure: https://cdash.dealii.org/test/22159767
```
*** The MPI_Comm_rank() function was called after MPI_FINALIZE was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
[testsuite:1942435] Local abort after MPI_FINALIZE started completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```

Comment (bangerth) [reply tests/gmsh/gmsh_api_01.cc:28]: I see. Thanks for the clarification!

**Review by bangerth — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]
